sqlglot/optimizer/simplify.py

```diff
@@ -167,89 +77,22 @@ def wrapped(expression, *args, **kwargs):
     return decorator
 
 
-def rewrite_between(expression: exp.Expression) -> exp.Expression:
-    """Rewrite x between y and z to x >= y AND x <= z.
-
-    This is done because comparison simplification is only done on lt/lte/gt/gte.
-    """
-    if isinstance(expression, exp.Between):
-        negate = isinstance(expression.parent, exp.Not)
+def annotate_types_on_change(func):
+    @wraps(func)
+    def _func(self, expression: exp.Expression, *args, **kwargs) -> t.Optional[exp.Expression]:
+        new_expression = func(self, expression, *args, **kwargs)
 
-        expression = exp.and_(
-            exp.GTE(this=expression.this.copy(), expression=expression.args["low"]),
-            exp.LTE(this=expression.this.copy(), expression=expression.args["high"]),
-            copy=False,
-        )
+        if new_expression is None:
+            return new_expression
 
-        if negate:
-            expression = exp.paren(expression, copy=False)
-
-    return expression
-
-
-COMPLEMENT_COMPARISONS = {
-    exp.LT: exp.GTE,
-    exp.GT: exp.LTE,
-    exp.LTE: exp.GT,
-    exp.GTE: exp.LT,
-    exp.EQ: exp.NEQ,
-    exp.NEQ: exp.EQ,
-}
+        if new_expression != expression:
+            self._annotator.clean()
+            new_expression = self._annotator.annotate(
+                expression=new_expression, annotate_scope=False
+            )
```

Comment:
@geooo109 I thought we were keeping the old type to avoid changing schemas after simplifying, right?